requirements.txt

```diff
@@ -1,6 +1,40 @@
-zimscraperlib>=2.1.0,<2.2
-kiwixstorage>=0.8.3,<0.9
-jinja2>=3.1.2<3.2
+## Direct dependencies
+zimscraperlib==3.1.0
+kiwixstorage==0.8.3
+Jinja2==3.1.2
 pif==0.8.2
-beautifulsoup4>=4.9.0,<5.0
-retrying>=1.3.4,<1.4
+beautifulsoup4==4.9.3
+retrying==1.3.4
+
+## The following requirements were added by pip freeze:
```

Comment:
@rgaudin is it correct, we have to pin all dependencies like this ?

Reply:
No;
Well that's debatable but the strategy behind [the doc](https://github.com/openzim/_python-bootstrap/wiki/pyproject.toml#dependencies) (I just added one line) is to only care about our own, direct dependencies.

We select them with care and when one of our dependency needs special attention (having a sub-dependency issue or using a range that we're afraid of), we pin the sub-dependency, with a comment stating why it's present.

`zimscraperlib` comes with many dependencies for instance. We only mention (and pin) it and it will pull `requests` (along with others). As soon as you'll be using `requests`, it will become a direct dependency and you'll pin it.

It doesn't require too much maintenance overhead and responsibilities are clear. PyPI prevents overwriting a version so there's no risk of change.
The risk lies in our direct dependencies definition. If they use ranges, there's a possibility for one of their dependency to change.

This is mitigated by:
- carefully choosing direct dependencies that care about their own.
- allowing ourselves to add and pin sub-dependencies
- tests whenever we can
- publishing docker images of all our releases.

I think it's a good compromise and should the need for more control arise (we've only add a couple issues with our wide ranges in years), we can still go beyond and freeze dependencies during the build process (thanks to hatch's flexibility, it should be easy) and/or use this lock file.

Reply:
I had doubts. You convinced me ;)